Automattic/wp-calypso
# Login: Display an error on the 2FA code component when the user enters an invalid code

<img width="416" alt="screen shot 2017-05-03 at 4 32 47 pm" src="https://cloud.githubusercontent.com/assets/1130674/25685699/288ac596-301e-11e7-938c-4bced6933fd5.png">

This PR updates `VerificationCodeForm` to:

- Display an error when the user enters an invalid code.
- Disable the submit button when the request is loading.
- Reset the `two_step_nonce` in memory to the new nonce returned by the API when an invalid code is entered.

**Testing**

- Visit the [`Login` page](http://calypso.localhost:3000/login) in incognito
- Submit a valid username/password for an account that has 2FA enabled and either SMS or the Authenticator app as its primary method of auth
- Submit an empty verification code
- Assert that you see an error message
- Submit an invalid verification code
- Assert that you see the error message like above
- Submit a valid verification code
- Assert that you are redirected to the Reader and logged in
